flang/lib/Lower/CallInterface.cpp

```diff
@@ -597,3 +597,32 @@ mlir::FunctionType Fortran::lower::translateSignature(
   return SignatureBuilder{characteristics.value(), converter, forceImplicit}
       .getFunctionType();
 }
+
+mlir::FuncOp Fortran::lower::declareFunction(
+    const Fortran::evaluate::ProcedureDesignator &proc,
+    Fortran::lower::AbstractConverter &converter,
+    llvm::Optional<llvm::StringRef> name) {
+
+  const auto *symbol = proc.GetSymbol();
+  assert(symbol && "declareFunction must be used with user defined procedures");
+  if (!name) {
+    auto name = converter.mangleName(*symbol);
```

Comment:
bug?

Reply:
No, I had decided to make name an optional argument (llvm::Optional<llvm::StringRef>), but that is not useful (no use case yet and  bit weird). I made `name` mandatory and removed this.